common/persistence/serialization/task_serializer.go

```diff
@@ -817,13 +817,18 @@ func (s *TaskSerializer) timerWorkflowCleanupTaskToProto(
 		TaskId:              workflowCleanupTimer.TaskID,
 		VisibilityTime:      &workflowCleanupTimer.VisibilityTimestamp,
 		BranchToken:         workflowCleanupTimer.BranchToken,
+		TaskDetails: &persistencespb.TimerTaskInfo_DeleteHistoryEventTaskDetails_{
+			DeleteHistoryEventTaskDetails: &persistencespb.TimerTaskInfo_DeleteHistoryEventTaskDetails{
+				AlreadyArchived: workflowCleanupTimer.WorkflowDataAlreadyArchived,
```

Comment:
set the default value to true?

Reply:
If this task wasn't made by the archival queue, that means the history hasn't actually been archived yet, so we can't set this flag to true.